corehq/apps/app_manager/views.py

```diff
@@ -2049,6 +2050,8 @@ def edit_advanced_form_actions(request, domain, app_id, module_id, form_id):
     json_loads = json.loads(request.POST.get('actions'))
     actions = AdvancedFormActions.wrap(json_loads)
     form.actions = actions
+    if advanced_actions_use_usercase(form.actions) and not is_usercase_in_use(domain):
```

Comment:
You're right @snopoke. I'm happy to take it out of this function and `edit_form_actions()`. I'm wondering though, in `enable_usercase()` the `is_usercase_in_use` check is done inside a `CriticalSection`. Is that much of a performance hit? Should we be checking whether we need to call `enable_usercase()` first, or is it fine just to call it, and return if usercase is already enabled.